sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASAuthDetails,
+  CASSupplierResponse,
+} from "./models/cas-supplier-response.model";
+import { AxiosRequestConfig } from "axios";
+import { HttpService } from "@nestjs/axios";
+import { CASIntegrationConfig, ConfigService } from "@sims/utilities/config";
+import { stringify } from "querystring";
+import { CustomNamedError } from "@sims/utilities";
+import { CAS_AUTH_ERROR } from "@sims/integrations/constants";
+
+@Injectable()
+export class CASService {
+  private readonly casIntegrationConfig: CASIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly httpService: HttpService,
+  ) {
+    this.casIntegrationConfig = config.casIntegration;
+  }
+
+  /**
+   * Request to login on CAS API and return CAS auth details with the token used for authentication in all other requests.
+   * @returns CAS auth details.
+   */
+  async logon(): Promise<CASAuthDetails> {
+    const url = `${this.casIntegrationConfig.baseUrl}/oauth/token`;
+    const auth = Buffer.from(
+      `${this.casIntegrationConfig.clientCredential.clientId}:${this.casIntegrationConfig.clientCredential.clientSecret}`,
+    ).toString("base64");
+    const headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Basic ${auth}`,
+    };
+    const body = { grant_type: "client_credentials" };
+    const config: AxiosRequestConfig = {
+      headers,
+    };
+    const data = stringify(body);
+    try {
+      return await this.httpService.axiosRef.post(url, data, config);
+    } catch (error: unknown) {
+      throw new CustomNamedError(
```

Comment:
I believe we should add a `logger.error` otherwise we may have no idea what caused the issue.